integration_tests/src/main/python/date_time_test.py

```diff
@@ -215,6 +215,21 @@ def test_to_unix_timestamp(data_gen, ansi_enabled):
         {'spark.sql.ansi.enabled': ansi_enabled})
 
 
+@pytest.mark.parametrize('time_zone', ["UTC", "UTC+0", "UTC-0", "GMT", "GMT+0", "GMT-0"], ids=idfn)
```

Comment:
looks like we need a test for the NULL timezone.

Reply:
This will be a bit tricky to test since when Spark sees the NULL literal it will optimize out the expression entirely because `FromUTCTimestamp` is `NullIntolerant`.

Reply:
can be an extra test with a null timezone column then or a udf

Reply:
The time zone here must be a literal string instead of a column.

Reply:
> can be an extra test with a null timezone column then or a udf

That's not going to work because timezone columns are not supported and a UDF will keep the entire ProjectExec off the GPU because we can't translate the UDF child expression to the GPU.